docs/details/algorithm.dox

```diff
@@ -107,9 +107,20 @@ Return type is u32 for all input types
 
 \ingroup scan_mat
 
-Perform inclusive sum along specified dimension
+Compute the cumulative sum (inclusive) along the specified dimension
 
-This table defines the return value types for the corresponding input types
+Here is a simple example for 1D arrays:
+
+\snippet test/scan.cpp ex_accum_1D
+
+For 2D arrays (and higher dimensions), you can specify the dimension along which
+the cumulative sum will be performed. If none is specified, it will be performed
+along the first dimension (0):
```

Comment:
Can you please verify if it dimension defaults to 0 or the first non-singleton dimension.

Reply:
Sounds good. 

PS. Defaults are only available for C++ API.

Reply:
This looks like a bug

Reply:
Were you expecting the default to be `-1` ? I was, but the code has `0` as default right now for whatever reason.

Reply:
I expect it to work on the first non-zero dimension. Does -1 do that? If it does then yes it should be the default. If it doesn't then it should be implemented.

Reply:
It should be fixed in a different PR